frontend/src/app/services/events.ts

```diff
@@ -1,32 +1,137 @@
-import {cloneDeep, get, isNil, shuffle} from 'lodash';
-import {eventList, Event} from './event-list';
-import {DayState} from './simulation';
+import {cloneDeep, get, isNil, last, shuffle, sample} from 'lodash';
+import {eventTriggers} from './event-list';
+import {DayState, MitigationEffect} from './simulation';
 
-interface TriggeredEvents extends Event {
-  triggered: boolean;
+// infinite timeout (1 milion years)
+const infTimeout = 1_000_000 * 365;
+
+export interface EventMitigation extends MitigationEffect {
+  timeout: number;
+  label: string;
+  id?: string;
+}
+
+export interface Event {
+  title: string;
+  text: string;
+  help?: string;
+  mitigations: EventMitigation[];
+}
+
+interface EventDef {
+  title: string;
+  text: string;
+  help?: string;
+  mitigations: Partial<EventMitigation>[];
```

Comment:
Mitigations have too many parameters, we do not require all of them in the definitions.